api/db/changelog/db.changelog-73-add-BillingProjectBufferEntry-table.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="ericsong" id="db.changelog-73-add-BillingProjectBufferEntry-table">
+    <createTable tableName="BillingProjectBufferEntry">
+
+      <column name="id" type="bigint" autoIncrement="true">
+        <constraints primaryKey="true" nullable="false" />
+      </column>
+
+      <column name="project_name" type="varchar(255)">
+        <constraints unique="true" nullable="false" />
+      </column>
+
+      <column name="creation_time" type="datetime">
+        <constraints nullable="false" />
+      </column>
+
+      <column name="status" type="varchar(32)">
```

Comment:
Made the change. But I am wondering... how much flexibility does this really give us in being able to change our enum values? If we expose the enum string values as part of our API, we're still unable to rename values without a major API version upgrade.

On the downside, this makes analyzing our data through SQL a little bit harder. And SQL that we do write needs to use short values like the one in my Dao, `SELECT COUNT(*) FROM BillingProjectBufferEntry WHERE status=0 OR status=2 OR status=3`.

Do you have a suggestion for being able to do the above without having to use the short values?

Reply:
The guiding principle for me on this subject is to avoid situations where your storage implementation restricts your ability to change an enum name in code for clarity. APIs I think are a separate question - API definitions should generally be decoupled from storage representations as a design principle; you need to maintain flexibility to change these independently.

Agreed this makes raw SQL interactions less readable; I'd guess there are tricks for dealing with this but I'm not aware of any easy fix. In that particular case, one alternative is to paramterize the set of statuses rather than encoding them directly in the Dao, and pass that in as a `Set<Short>`, `WHERE status IN @statuses`

Reply:
Also, more context on that decision here: https://precisionmedicineinitiative.atlassian.net/browse/RW-872

I think it's reasonable to disagree on preferred approach here, both have merit. Since we're already using this approach, I would definitely prefer to keep that current pattern unless I am overwhelmingly overruled by the rest of the team.

Reply:
I am cool with following the current paradigm.